models/comments-models.js

```diff
@@ -1,9 +1,23 @@
 const connection = require("../db/connection")
-const { checkReviewExists } = require("./reviews-models")
+const { checkReviewExists, getAuthor } = require("./reviews-models")
 
 exports.fetchComments = (review_id) => {
     return checkReviewExists(review_id).then(()=>
     {return connection.query(`SELECT * FROM comments WHERE review_id=$1 ORDER BY created_at DESC;`,[review_id]).then((result)=> {
         return result.rows
     })})
+}
+
+exports.insertComment = (comment, review_id) => {
+    const intID = Number(review_id)
+    if(comment.body && comment.username){
+    const {username, body} = comment
+    const newCommentQuery = `INSERT INTO comments (body, author, votes, review_id, created_at) VALUES ($1, $2, $3, $4, NOW()) RETURNING *`
+    return getAuthor(review_id).then((author) => {
+        return connection.query(newCommentQuery,[body, author,0,intID]).then((result)=>{
```

Comment:
the author associated with the comment should be the user submitting the comment, so the username in the sent comment object, not the author of the article.